lib/linguist/generated.rb

```diff
@@ -314,7 +314,7 @@ def generated_protocol_buffer?
       return false unless PROTOBUF_EXTENSIONS.include?(extname)
       return false unless lines.count > 1
 
-      return lines[0].include?("Generated by the protocol buffer compiler.  DO NOT EDIT!")
+      return lines.first(6).any? { |l| l.include?("Generated by the protocol buffer compiler.  DO NOT EDIT!") }
```

Comment:
I'm having trouble finding examples where this warning statement is further down than the 3rd line. Maybe change to the following?
```suggestion
      return lines.first(3).any? { |l| l.include?("Generated by the protocol buffer compiler.  DO NOT EDIT!") }
```